include/boost/geometry/algorithms/detail/overlay/get_turns.hpp

```diff
@@ -161,15 +162,16 @@ private :
         if (! m_point_retrieved)
         {
             advance_to_non_duplicate_next(m_current_point, m_circular_iterator);
-            m_point = *m_circular_iterator;
+            m_point = boost::addressof(*m_circular_iterator);
```

Comment:
Actually it's copy assignment. Are you sure we don't require it? Is it tested anywhere? Where does it fail?

Reply:
Right, copy assignment. So it does not work with certain point types (such as the one I provided).

Now I see your question, I'm considering why we can't pass the result of the iterator anyway, without a point. I'll check it later.

Reply:
Fixed, I return the iterator now.
That point is retrieved lazily, because often it's not necessary. Only when asked for the circular iterator moves forward to the point, that happens only once. Then it can just return that point (as also happens later in that same header file).

Reply:
Right, this is better.

I'm wondering which point types didn't work with the previous version? It seems wierd to me that we allow points that cannot be copied or assigned to. Esspecially in functions that work for containers/ranges and in order to store a point in a container one has to copy or move assign. So I have a problem with figuring out the case we talk about here.

Reply:
This class has implicitly-declared copy and move assignment operators and constructors. There is nothing preventing assignment here. There are only no setters for coordinates. E.g. this compiles:
```
const_point pt(0, 0);
const_point pt2 = pt;
const_point pt3 = std::move(pt);
pt2 = pt;
pt3 = std::move(pt);
```